### ISSUE TITLE
object-curly-spacing not working for union types

### ISSUE THREAD

**Issue by benwaffle** [palantir/tslint#4385]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.2.2
- __Running TSLint via__: (pick one) CLI / Node.js API / VSCode / grunt-tslint / Atom / Visual Studio / etc: CLI

#### TypeScript code being linted

```ts
async function updateUser(req: Request & { file: Express.MulterS3.File}, res: Response) {
```

with `tslint.json` configuration:

```json
{
    "defaultSeverity": "error",
    "extends": [
        "tslint:recommended",
        "tslint-config-airbnb"
    ],
    "jsRules": {},
    "rules": {
        "ter-indent": [true, 4],
        "object-literal-sort-keys": false,
        "no-console": false,
        "object-shorthand-properties-first": false
    },
    "rulesDirectory": [
        "node_modules/tslint-eslint-rules/dist/rules"
    ]
}
```

#### Actual behavior
No warnings are printed 

#### Expected behavior
object-curly-spacing should warn about `{ file: Express.MulterS3.File}`

**Comment by benwaffle** [comment]
(let me know if you cannot reproduce, i will provide the entire file)

**Comment by JoshuaKGoldberg** [comment]
@benwaffle `object-curly-spacing` is a rule added by [`tslint-eslint-rules`](https://github.com/buzinas/tslint-eslint-rules/blob/master/src/docs/rules/objectCurlySpacingRule.md). You'll want to file an issue on them.

**Comment by benwaffle** [comment]
My bad. Doesn't tslint have it's own version of that warning?
